src/Microsoft.Azure.SignalR.Management/ServiceManagerOptions.cs

```diff
@@ -36,40 +34,73 @@ public class ServiceManagerOptions
         /// <summary>
         /// Gets or sets the service endpoint for accessing Azure SignalR Service.
         /// </summary>
-        public ServiceEndpoint ServiceEndpoint
+        public ServiceEndpoint ServiceEndpoint { get; set; }
+
+        /// <summary>
+        /// Gets or sets the service endpoints for accessing Azure SignalR Service and switches to multi-endpoint mode.
+        /// </summary>
+        internal ServiceEndpoint[] ServiceEndpoints { get; set; } //not ready for public use
+
+        /// <summary>
+        /// Gets or sets the transport type to Azure SignalR Service. Default value is Transient.
+        /// </summary>
+        public ServiceTransportType ServiceTransportType { get; set; } = ServiceTransportType.Transient;
+
+        /// <summary>
+        /// A unified way to get the service endpoint(s) set by "ConnectionString", "ServiceEndpoint" or "ServiceEndpoints".
+        /// </summary>
+        internal ServiceEndpoint[] UnifiedEndpoints
```

Comment:
no need to add a property, leverage `ServiceEndpoints`? also do the validation once instead of every `get`